### PR TITLE
EVG-15836 Fix s3_copy with fixed smoke test

### PR DESCRIPTION
**Pull request by malikchaya2**
[EVG-15836](https://jira.mongodb.org/browse/EVG-15836)

### Description 
This includes the previously merged and reverted s3copy fix, as well as a fix for the smoke test (the second commit). 

### Testing 
patch: https://spruce.mongodb.com/version/61dc5fdf850e6134cad323fc/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

### CODE REVIEW

File: agent/testdata/agent-integration.yml [hunk 1]
```diff
@@ -299,10 +299,10 @@ tasks:
         s3_copy_files:
         - source:
             bucket: build-push-testing
-            path: pushtest-stage/unittest-${task_id}-${distro_exp}-${buildvar_exp}-${file_exp}.txt
+            path: unittest-${distro_exp}-${buildvar_exp}.txt
```
Comment (Kimchelly) [inline agent/testdata/agent-integration.yml:302]: Is changing this still necessary if the above retry loop was changed? It seems like the original s3.put/s3Copy.copy logic should work as intended if the above worked.

**Review by Kimchelly — CHANGES_REQUESTED** [review]

File: agent/testdata/agent-integration.yml (hunk 1 above)
Comment (malikchaya2) [reply agent/testdata/agent-integration.yml:302]: no, I thought I removed that. I'll remove.

Comment (malikchaya2) [reply agent/testdata/agent-integration.yml]: fixed

**Review by ablack12 — APPROVED** [review]

**Review by Kimchelly — APPROVED** [review]
LGTM mod the smoke test passing.

**Pull request merged**